bootstrap/config.go

```diff
@@ -131,22 +134,38 @@ func (c *Config) ReadFromEnvironment(environ *env.Environment) map[string]string
 	changed := map[string]string{}
 
 	// Use reflection for the type and values
-	t := reflect.TypeOf(*c)
-	v := reflect.ValueOf(c).Elem()
+	fields := reflect.TypeOf(*c)
+	values := reflect.ValueOf(c).Elem()
 
 	// Iterate over all available fields and read the tag value
-	for i := 0; i < t.NumField(); i++ {
-		field := t.Field(i)
-		value := v.Field(i)
+	for i := 0; i < fields.NumField(); i++ {
+		f := fields.Field(i)
+		v := values.Field(i)
 
 		// Find struct fields with env tag
-		if tag := field.Tag.Get("env"); tag != "" && environ.Exists(tag) {
-			newValue, _ := environ.Get(tag)
-
-			// We only care if the value has changed
-			if newValue != value.String() {
-				value.SetString(newValue)
-				changed[tag] = newValue
+		if tag := f.Tag.Get("env"); tag != "" && environ.Exists(tag) {
+			newStr, _ := environ.Get(tag)
+
+			switch v.Kind() {
+			case reflect.String:
+				if newStr == v.String() {
+					break
+				}
+				v.SetString(newStr)
+				changed[tag] = newStr
+			case reflect.Bool:
+				newBool, err := strconv.ParseBool(newStr)
+				if err != nil {
+					log.Printf("warning: cannot parse %s=%s as bool, ignoring", tag, newStr)
+					break
+				}
+				if newBool == v.Bool() {
+					break
+				}
+				v.SetBool(newBool)
+				changed[tag] = newStr
+			default:
+				log.Printf("warning: bootstrap.Config.ReadFromEnvironment does not support %v for %s", v.Kind(), tag)
```

Comment:
The only other type in the Config struct is `int`, but it doesn't have an `env` tag so I don't think it's relevant here yet. If it becomes relevant, we can add a `case reflect.Int` with int-parsing etc.

I felt like a warning was better than an error… maybe 🤔